### PR TITLE
Add cache.ttl to add_host_metadata

### PR DESCRIPTION
**Pull request by cwurm**
Despite what is says in the documentation (`NOTE: The host information is refreshed every 5 minutes.`) the `add_host_metadata` processor does not actually ever refresh most of its information (it does the IPs and MACs if `netinfo.enabled: true`, but not the base host info).

This PR changes it to call `sysinfo.Host()` every time its cache expires.

It also adds a `cache.ttl` parameter to configure how often that happens. The default stays at `5m`, and negative values disable caching altogether (most likely at some cost to performance).

I did not add the new config parameter to all the `*beat.reference.yml` files - I noticed they didn't contain everything (e.g. `add_kubernetes_metadata`) and I think this parameter is not something most users need to configure. Happy to put it in though if it makes sense.

### CODE REVIEW

File: libbeat/processors/add_host_metadata/add_host_metadata.go [hunk 1]
```diff
@@ -75,28 +67,41 @@ func newHostMetadataProcessor(cfg *common.Config) (processors.Processor, error)
...
-		return
+		return nil
+	}
+
+	h, err := sysinfo.Host()
+	if err != nil {
+		return err
 	}
 
-	data := host.MapHostInfo(p.info)
```
Comment (andrewkroh) [inline libbeat/processors/add_host_metadata/add_host_metadata.go]: IIUC the problem was here b/c it never fetched updated info

**Comment by ruflin** [comment]
I added `needs_backport` as I think we should backport this fix.

File: libbeat/processors/add_host_metadata/add_host_metadata.go (hunk 1 above)
Comment (cwurm) [reply libbeat/processors/add_host_metadata/add_host_metadata.go]: Correct

**Comment by cwurm** [comment]
I've added two changelog entries (one under Bugfix, one under Added) - let me know if it's good to merge please.

**Pull request merged**